### DISCUSSION TITLE
Webplayer scrubbing and shuffle?

### DISCUSSION THREAD

**Discussion by troycarpenter** [ampache/ampache#4002] [Q&A] [answered]
I like the way the webplayer is shaping up, but it seems to have lost the shuffle functionality (keep playing the current song and shuffle all others after it).

I may have asked this before, but is there any way to scrub through the song?  Anything I do either tries to pick up elements, or it restarts the song from the beginning.

(It took me a short minute to figure out that turning on the loop playlist turns off the remove played songs.  Nice.  I also like that the visualizer works since the last time I looked.)

**Comment by lachlan-00** [comment]
that's annoying, not sure how i managed to screw up shuffle, it's been a while since i did anythign with it so i'll see what's going on there.

when you say scrub a song you mean select different times on the bar? there can be issues with transcoding and getting selecting different times but i'm not too sure.

**Reply by troycarpenter** [reply]
Yes for the scrubbing.  I should be able to grab the location marker and move it back and forth on the song.  But basically I don't seem to have a way to go to an arbitrary point in a song.  Most players seem to have the ability to skip around in a song using some kind of playback marker.

**Reply by lachlan-00** [reply]
what format is your output stream? the usual culprit is that it hasn't loaded the full length of the file or the stream output doesn't like seeking

**Reply by troycarpenter** [reply]
Have a look at my other complaint about the wrong song durations.  These are mp3 files, but something is definitely processing them incorrectly for playback.

**Reply by lachlan-00** [reply]
do you get a duration with your tracks? they should be sent using the duration property now (check the JS console)
![image](https://github.com/user-attachments/assets/cee80775-7f72-4557-b9cd-18256e905071)

**Reply by troycarpenter** [reply]
Yes, it appears the duration value is correct:
![image](https://github.com/user-attachments/assets/bb542b72-2a20-48fe-96e5-e8e4af9ffe95)
However, the web player is still getting it wrong:
![image](https://github.com/user-attachments/assets/5c4d4321-bc6a-47da-838c-db04595c5bfa)
This same song is only playing to 3:01, not 3:23 as expected.

**Comment by lachlan-00** [comment]
![image](https://github.com/user-attachments/assets/c821c090-8388-46aa-9175-89e26aa0a1ae)
js bug so that's why it refreshes the original list

**Comment by troycarpenter** [comment, marked as answer]
The problem described arose from this setting in the configuration file:
```
; Waveform
; This settings tells Ampache to attempt to generate a waveform
; for each song. It requires transcode and encode_args_wav settings.
; You must also set tmp_dir_path in order for this to work
; DEFAULT: "false"
waveform = "false"
```
When that is set to true, it forces transcoding and apparently that transcoded song is sent to the player instead of the original.  This transcoded song has the wrong durations and messes with what the players think the duration actually is.

By turning this setting off, and deleting any player cache that may exist, the song duration problem is fixed.  As a result, the web player no longer shows an incorrect wave form, but instead a scrub bar which can be used to randomly search through the song.
![image](https://github.com/user-attachments/assets/4f59fedf-9e67-4006-b873-d74564071ee7)
You can see in the screenshot that the waveform was replaced with a simple scrub bar and the duration of the song is now correct.  This is the same issue for #4001.

The shuffle feature is still MIA, but I'm sure that will be fixed before long.
